File: src/main/java/com/example/testInterface/controller/LightingAPI.java
package com.example.testInterface.controller;

import com.example.testInterface.entity.Lighting;
import com.example.testInterface.service.ILightingService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.AutoConfiguration;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Controller
@RequestMapping("/api/")
public class LightingAPI {
    @Autowired
    ILightingService lightingService;
    @Async
    @GetMapping("/main")
    public ResponseEntity<List<Lighting>> getAllLighting(){
        System.out.println("Get Lighting list for main page");
        return new ResponseEntity<>(lightingService.getList(), HttpStatus.OK);
    }

    @Async
    @PostMapping("/create")
    public List<Lighting> createNew(){
        return lightingService.getList();
    }

    @Async
    @GetMapping("/edit/{id}")
    public Lighting getForEdit(@PathVariable("id") String id){
        return lightingService.getLightingID(id);
    }

    @Async
    @PostMapping("/edit/{id}")
    public ResponseEntity<?> saveEdit(@RequestBody Lighting lighting){
        lightingService.update(lighting);
        return new ResponseEntity<>(HttpStatus.CREATED);
    }
    @Async
    @DeleteMapping("/delete/{id}")
    public void delete(@PathVariable("id") String id){
        lightingService.deleteLighting(id);
    }
    @Async
    @GetMapping("/info/switchLight/{id}")
    public void switchLight(@PathVariable("id") String id) {
        lightingService.switchLight(id);
    }
}

File: frontend/src/component/LightingAPI.js
import React, { useState, useEffect } from 'react';
import axios from 'axios';


function LightingAPI(){
  const [lightingList, setLightingList] = useState([]);

  useEffect(() => {
    axios.get('/api/main')
      .then(response => {
        setLightingList(response.data);
      })
      .catch(error => {
        console.log(error);
      });
  }, []);

  const createNew = () => {
    axios.post('/api/create')
      .then(response => {
        setLightingList(response.data);
      })
      .catch(error => {
        console.log(error);
      });
  };

  const getForEdit = (id) => {
    axios.get(`/api/edit/${id}`)
      .then(response => {
        // handle the response
      })
      .catch(error => {
        console.log(error);
      });
  };

  const saveEdit = (lighting) => {
    axios.post(`/api/edit/${lighting.id}`, lighting)
      .then(response => {
        // handle the response
      })
      .catch(error => {
        console.log(error);
      });
  };

  const deleteLighting = (id) => {
    axios.delete(`/api/delete/${id}`)
      .then(response => {
        setLightingList(lightingList.filter(lighting => lighting.id !== id));
      })
      .catch(error => {
        console.log(error);
      });
  };

  const switchLight = (id) => {
    axios.get(`/api/info/switchLight/${id}`)
      .then(response => {
        // handle the response
      })
      .catch(error => {
        console.log(error);
      });
  };
}

export default LightingAPI;